sparse/tests/test_core.py

```diff
@@ -136,6 +137,10 @@ def test_dot():
     assert_eq(a.dot(b), sa.dot(sb))
     assert_eq(np.dot(a, b), sparse.dot(sa, sb))
 
+    if sys.version_info >= (3, 5):
+        assert_eq(eval("a @ b"), eval("sa @ sb"))
+        assert_eq(eval("sa @ sb"), sparse.dot(sa, sb))
```

Comment:
I don't think this should be merged unless ``x @ M`` for an array ``x`` and a matrix ``M`` works correctly. There's no reason why it shouldn't.